### PR TITLE
Update citing info with user manual

### PR DESCRIPTION
**Pull request by valeriabarra**
This PR adds the bib file for libCEED's user manual and updates this info on our "How to Cite" section of the README. 

Tentatively, I replaced the dev-site info that was sitting there and does not have a DOI (to avoid the spread of possible citation modes mentioned in our recent conversation). But if this is not desired, I can put that piece back. 

Also while building the docs, I stumbled upon a small linking bug in the `v0.7` section of the release notes and fixed it.

### CODE REVIEW

**Review by jedbrown — APPROVED** [review]

File: README.rst [hunk 1]
```diff
@@ -429,11 +429,27 @@ How to Cite
...
-     howpublished = {\url{https://github.com/ceed/libceed}},
-     year = 2020
+   @misc{libceed-user-manual,
+     author       = {Abdelfattah, Ahmad and
+                     Barra, Valeria and
+                     Beams, Natalie and
+                     Brown, Jed and
+                     Camier, Jean-Sylvain and
+                     Dobrev, Veselin and
+                     Jean-Sylvain, Yohann and
```
Comment (LeilaGhaffari) [inline README.rst:439]: Is this name correct? Isn't it Yohann Dodouit?

Comment (tzanio) [inline README.rst:439]:
```suggestion
                     Dudouit, Yohann and
```

Comment (valeriabarra) [reply README.rst:439]: Thanks for the good catch! 

Weird this happened though. I downloaded the `bib` file as it is automatically generated on Zenodo.

@jedbrown was this list automatically generated from the [AUTHORS](https://github.com/CEED/libCEED/blob/main/AUTHORS) file? Or was this using the script for uploading to Zenodo?

Comment (valeriabarra) [reply README.rst:439]: Thanks! I fixed it here. Although the problem remains on [Zenodo])(https://zenodo.org/record/4302737/export/hx#.X8lEuVlKgUE) at the moment.

**Comment by jedbrown** [comment]
Fixed in the record now.

**Comment by valeriabarra** [comment]
> Fixed in the record now.

Thanks! 
I'll go ahead and merge this.

**Comment by valeriabarra** [comment]
This can't be merged at the moment because the required test with Intel compilers fails, although unrelated to this PR.

@jeremylt are you aware of the issue?

**Comment by jeremylt** [comment]
Yep. I have a PR with a fix open.

**Comment by jeremylt** [comment]
Well, something new is broken with ifort. Intel is a never-ending source of debugging fun.

**Comment by valeriabarra** [comment]
Oh! 
I just rebased onto main, by the way.

**Comment by jeremylt** [comment]
@jedbrown do we want to just dump icc/ifort in CI until they can make an official action?

https://github.com/oneapi-src/oneapi-ci/issues/31

**Comment by jedbrown** [comment]
Seems like a bug we can report upstream and pin the version to the version last time our tests were passing.
```
ifort: error #10105: ld: core dumped
Makefile:453: recipe for target 'build/t000-ceed-f' failed
ifort: warning #10102: unknown signal(-1822799696)
ifort: error #10106: Fatal error in ld, terminated by unknown
make: *** [build/t000-ceed-f] Error 1
make: *** Waiting for unfinished jobs....
ifort: error #10105: ld: core dumped
ifort: warning #10102: unknown signal(1202989232)
ifort: error #10106: Fatal error in ld, terminated by unknown
make: *** [build/t001-ceed-f] Error 1
Makefile:453: recipe for target 'build/t001-ceed-f' failed
Error: Process completed with exit code 2.
```

**Review by jedbrown — APPROVED** [review]

**Pull request merged**